terraform/groups/ecs-service/locals.tf

```diff
@@ -9,39 +9,15 @@ locals {
   lb_listener_paths         = ["/accounts-filing*"]
   healthcheck_path          = "/accounts-filing/healthcheck" #healthcheck path for accounts-filing-web
   healthcheck_matcher       = "200"                        # no explicit healthcheck in this service yet, change this when added!
-
+  application_subnet_ids     = data.aws_subnets.application.ids
   kms_alias       = "alias/${var.aws_profile}/environment-services-kms"
   service_secrets = jsondecode(data.vault_generic_secret.service_secrets.data_json)
-
-  parameter_store_secrets = {
-    "vpc_name"             = local.service_secrets["vpc_name"]
-    "chs_api_key"          = local.service_secrets["chs_api_key"]
-    "internal_api_url"     = local.service_secrets["internal_api_url"]
-    "chs_internal_api_key" = local.service_secrets["chs_internal_api_key"]
-    "cache_server"         = local.service_secrets["cache_server"]
-    "account_url"          = local.service_secrets["account_url"]
-    "cookie_secret"        = local.service_secrets["cookie_secret"]
-    "oauth2_auth_uri"      = local.service_secrets["oauth2_auth_uri"]
-    "oauth2_redirect_uri"  = local.service_secrets["oauth2_redirect_uri"]
-    "oauth2_token_uri"     = local.service_secrets["oauth2_token_uri"]
-    "oauth2_client_id"     = local.service_secrets["oauth2_client_id"]
-    "oauth2_client_secret" = local.service_secrets["oauth2_client_secret"]
-    "oauth2_request_key"   = local.service_secrets["oauth2_request_key"]
-  }
-
-  vpc_name             = local.service_secrets["vpc_name"]
-  chs_api_key          = local.service_secrets["chs_api_key"]
-  internal_api_url     = local.service_secrets["internal_api_url"]
-  chs_internal_api_key = local.service_secrets["chs_internal_api_key"]
-  cache_server         = local.service_secrets["cache_server"]
-  account_url          = local.service_secrets["account_url"]
-  cookie_secret        = local.service_secrets["cookie_secret"]
-  oauth2_auth_uri      = local.service_secrets["oauth2_auth_uri"]
-  oauth2_redirect_uri  = local.service_secrets["oauth2_redirect_uri"]
-  oauth2_token_uri     = local.service_secrets["oauth2_token_uri"]
-  oauth2_client_id     = local.service_secrets["oauth2_client_id"]
-  oauth2_client_secret = local.service_secrets["oauth2_client_secret"]
-  oauth2_request_key   = local.service_secrets["oauth2_request_key"]
+  stack_secrets              = jsondecode(data.vault_generic_secret.stack_secrets.data_json)
+  application_subnet_pattern = local.stack_secrets["application_subnet_pattern"]
+  use_set_environment_files  = var.use_set_environment_files
+  s3_config_bucket           = data.vault_generic_secret.shared_s3.data["config_bucket_name"]
+  app_environment_filename   = "accounts-filing-web.env"
+  vpc_name                   = data.aws_ssm_parameter.secret[format("/%s/%s", local.name_prefix, "vpc-name")].value
```

Comment:
We can simplify using 
  `vpc_name                  = local.stack_secrets["vpc_name"]`
here.